lms/djangoapps/verified_track_content/models.py

```diff
@@ -34,26 +34,39 @@ def move_to_verified_cohort(sender, instance, **kwargs):  # pylint: disable=unus
 
     if verified_cohort_enabled and (instance.mode != instance._old_mode):  # pylint: disable=protected-access
         if not is_course_cohorted(course_key):
-            log.error("Automatic verified cohorting enabled for course '%s', but course is not cohorted", course_key)
+            log.error("Automatic verified cohorting enabled for course '%s', but course is not cohorted.", course_key)
         else:
-            existing_cohorts = get_course_cohorts(get_course_by_id(course_key), CourseCohort.MANUAL)
-            if any(cohort.name == verified_cohort_name for cohort in existing_cohorts):
-                args = {
-                    'course_id': unicode(course_key),
-                    'user_id': instance.user.id,
-                    'verified_cohort_name': verified_cohort_name
-                }
-                # Do the update with a 3-second delay in hopes that the CourseEnrollment transaction has been
-                # completed before the celery task runs. We want a reasonably short delay in case the learner
-                # immediately goes to the courseware.
-                sync_cohort_with_mode.apply_async(kwargs=args, countdown=3)
-
-                # In case the transaction actually was not committed before the celery task runs,
-                # run it again after 5 minutes. If the first completed successfully, this task will be a no-op.
-                sync_cohort_with_mode.apply_async(kwargs=args, countdown=300)
+            course = get_course_by_id(course_key)
+            existing_manual_cohorts = get_course_cohorts(course, CourseCohort.MANUAL)
+            if any(cohort.name == verified_cohort_name for cohort in existing_manual_cohorts):
```

Comment:
@scottrish Note that we now have an inconsistency-- we require the course to have a single random cohort (for audit learners), but we allow multiple manual cohorts, as long as one of them has the special "verified" name as specified on the admin site.

We could simplify matters by also requiring that the course have a single manual cohort, which we would assume is for verified learners. This would eliminate the need to specify the verified cohort name in the admin site, which would be a big plus because course staff could rename that cohort without having to also update the name in the admin site (I'm afraid that course staff might rename the "verified" cohort without alerting the PM, causing the automatic cohorting feature to break).

What do you think? @dianakhuang has already merged the addition of the verified cohort name in the admin panel, but it would be simple to modify the code and delete the field from the model, since both migrations would be in the same release.